src/storage/pooled_storage_manager.h

```diff
@@ -32,6 +32,8 @@ class GPUPooledStorageManager final : public StorageManager {
    */
   GPUPooledStorageManager() {
     reserve_ = dmlc::GetEnv("MXNET_GPU_MEM_POOL_RESERVE", 5);
+    cudaGetDeviceCount(&ndev_);
```

Comment:
We don't call cuda in singleton constructors because cuda context cannot be shared after fork. This should be called on first use.